cmd/start.go

```diff
@@ -0,0 +1,238 @@
+			cmd.OutOrStdout(),
+		)
+		if err != nil {
+			return err
+		}
+	}
+
+	startParams := operations.NewStartWorkflowParams()
+	startParams.SetAccessToken(&o.token)
+	startParams.SetWorkflowIDOrName(o.workflow)
+	startParams.SetParameters(operations.StartWorkflowBody{
+		InputParameters:    o.parameters,
+		OperationalOptions: o.options,
+	})
+	startResp, err := api.Operations.StartWorkflow(startParams)
+	if err != nil {
+		return err
+	}
+
+	currentStatus := startResp.Payload.Status
+	statusMsg, err := workflows.StatusChangeMessage(o.workflow, currentStatus)
+	if err != nil {
+		return err
+	}
+	displayer.DisplayMessage(
+		statusMsg,
+		displayer.Success,
+		false,
+		cmd.OutOrStdout(),
+	)
+
+	if o.follow {
+		err = followWorkflowExecution(cmd, currentStatus, o.token, o.serverURL, o.workflow)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// validateStartOptionsAndParams gets the workflow parameters from the server and validates the options and params provided.
+// For operations options, it returns an error if any of them aren't valid. Translated options if necessary.
+// For input parameters, simply displays errors if any and continues execution.
+func validateStartOptionsAndParams(
+	api *client.API,
+	token, workflow string,
+	options, inputParams map[string]string,
+	out io.Writer,
+) (validatedOptions map[string]string, validatedParams map[string]string, err error) {
+	params := operations.NewGetWorkflowParametersParams()
+	params.SetAccessToken(&token)
+	params.SetWorkflowIDOrName(workflow)
+	paramsResp, err := api.Operations.GetWorkflowParameters(params)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	validatedOptions, err = validator.ValidateOperationalOptions(paramsResp.Payload.Type, options)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	validatedParams, errorList := validator.ValidateInputParameters(
+		inputParams,
+		paramsResp.Payload.Parameters,
+	)
+	for _, err := range errorList {
+		displayer.DisplayMessage(err.Error(), displayer.Error, false, out)
+	}
+	return validatedOptions, validatedParams, nil
+}
+
+// followWorkflowExecution follow the execution of the workflow, by calling the GetStatus endpoint periodically.
+// The interval used for the requests is dictated by config.CheckInterval.
+// If the workflow finishes successfully, this calls the ls command to display the workflow files' URLs.
+func followWorkflowExecution(
+	cmd *cobra.Command,
+	currentStatus string,
+	token, serverURL, workflow string,
+) error {
+	for slices.Contains([]string{"pending", "queued", "running"}, currentStatus) {
+		time.Sleep(time.Duration(config.CheckInterval) * time.Second)
+		status, err := workflows.GetStatus(token, workflow)
+		if err != nil {
+			return err
+		}
+		currentStatus = status.Status
+
+		statusMsg, err := workflows.StatusChangeMessage(workflow, currentStatus)
+		if err != nil {
+			return err
+		}
+		displayer.DisplayMessage(
+			statusMsg,
+			displayer.Success,
+			false,
+			cmd.OutOrStdout(),
+		)
```

Comment:
When the workflow fails, the client incorrectly shows `==> SUCCESS: workflow has been failed`. The python client has the same behaviour. What about opening an issue to fix this?

Reply:
I agree, I'll open the issues